### DISCUSSION TITLE
create debian package with version of npm 4.x.x

### DISCUSSION THREAD

**Discussion by maxlavr** [npm/feedback#248] [CLI] [answered]
Hello everyone. My name is Max and my profession is DevOps. 
I operate a product where the software has not been updated for a long time. This product use NPM with version 4.x.x. We need to build debian package from source code. Your project have officially repository in salsa.debian.org - https://salsa.debian.org/js-team/npm But there are no branch with name debian/4.x.x
Can you help me to find debian instructions for this version or maybe somebody have archives for me?

**Comment by MylesBorins** [comment, marked as answer]
@maxlavr thanks for reaching out. The debian npm ports are not maintained by the npm team themselves but rather by the ["Debian JavaScript Maintainers"](https://salsa.debian.org/js-team)

It is worth mentioning that npm 4.x.x is not maintained, nor has it been in quite some time. I'm unsure which version of Node.js you are using but I would warn that you are likely using a version that has security vulnerabilities.

There are debs that are maintained by Nodesource (a private company)
https://github.com/nodesource/distributions

It looks like Node.js 7.4.0 - 7.10.11 shipped with npm 4.x.x so you might be able to use their [7.x debs](https://github.com/nodesource/distributions/blob/master/deb/setup_7.x) to get the version of npm you are looking for. Again though I will warn that Node.js 7 and npm 4 have not been updated in close to 4 years and I would highly advise against using in production for security reasons

**Reply by maxlavr** [reply]
Thank you very much for your answer. I downloaded the Nodeource source archives and the debian source archive from https://deb.nodesource.com/node_7.x/pool/main/n/nodejs/
After I compiled and built myself a package with the necessary configurations, I installed it. The final package contains npm with the version of nodejs we are using, and we do not need to build it separately.
